### PR TITLE
test(HACBS-2312): add tests for collect-data task

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

**Comment by johnbieren** [comment]
Depends on https://github.com/redhat-appstudio/release-service-utils/pull/48

File: catalog/task/collect-data/0.1/tests/test-collect-data-with-extra-data.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+              metadata:
+                name: releaseplan-sample
+                namespace: default
+              spec:
+                application: foo
+                target: foo
+              EOF
+              kubectl apply -f releaseplan
+
+              cat > releaseplanadmission << EOF
```
Comment (mmalina) [inline catalog/task/collect-data/0.1/tests/test-collect-data-with-extra-data.yaml:47]: You could save the yaml in the workspace somewhere (a subdir maybe) and that way you could compare the whole `extraData` field directly rather then writing it again. Like this: https://mikefarah.gitbook.io/yq/usage/tips-and-tricks#comparing-yaml-files

But it's up to you.

File: catalog/task/collect-data/0.1/tests/test-collect-data-fail-missing-cr.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: test-collect-data-fail-missing-cr
+  annotations:
+    test/assert-task-failure: "run-task"
+spec:
+  description: |
+    Run the collect-data task without all required resources and verify that the task fails as expected.
```
Comment (mmalina) [inline catalog/task/collect-data/0.1/tests/test-collect-data-fail-missing-cr.yaml:10]: Can you specify here what's actually missing? I can't see it.

File: catalog/task/collect-data/0.1/tests/test-collect-data-with-extra-data.yaml (hunk 1 above)
Comment (mmalina) [reply catalog/task/collect-data/0.1/tests/test-collect-data-with-extra-data.yaml:47]: I see. But yeah, maybe it's not worth it - you already rewrote your `extraData` as json anyway and it's simple enough that this is probably fine for now.

**Review by mmalina — APPROVED** [review]

**Pull request merged**